### DISCUSSION TITLE
PyANSYS installation

### DISCUSSION THREAD

**Discussion by swparth** [ansys/pyansys#373] [Q&A] [answered]
Hi everyone,

I’m Parth Swaroop working in the simulations team in Applied Materials.

I want to install PYANSYS in the system(Windows 10) which has no internet access. If need be, then i can request for access (I will come to that later). 

I intend to create a document so that my team members can also follow it. Since there are many different type of Ansys users in group, I would like a one procedure fits all.

I found that there are online and offline options both for installation here :https://github.com/ansys/pyansys

My first attempt was to go for offline method. So what I really did is as below : 
1)Install Conda with admin rights and add it system environment variables. This, I was successful in doing. My C:\ProgramData\Anaconda3 has the python 3.11 package, with Jupyter, Spyder environment and so on. I intend to use Spyder to write my codes since I'm comfortable with that interface.  My path file has C:\ProgramData\Anaconda3\Script;C:\ProgramData\anaconda3;C:\ProgramData\anaconda3\Lib;

2) Install PyANSYS package. I have downloaded pyansys-v2023.2.9-all-wheelhouse-Windows-3.11 and extracted package. I used conda prompt to install the package. I started with ansys_api_fluent-0.3.12-py3-none-any.whl. It prompted me or showed the error  that I first need to install grpcio-1.56.2-cp311-cp311-win_amd64.whl, which i did. Then it asked for one more package before which i was able to install the above fluent-api. 

Do I have to install all .whl one by one or on need basis? Is there a simple way in which I can install entire package in one go?

3) If having the internet access is the easiest way then do let me know. However, I can access only specific pages which i request for. Can you point me to the list of URL from Ansys for which access is required during installation of PyAnsys? Subsequently can you tell the step to install them?

My knowledge installation using python is beginner/novice level.

**Comment by RobPasMue** [comment, marked as answer]
Hi @swparth - thanks a lot for opening this discussion! We are glad that you are considering the use of PyAnsys for your workflows.

If you have no internet connection on your machines, then we recommend that you use our wheelhouses (available in the [Releases](https://github.com/ansys/pyansys/releases/tag/latest) page of this repository). Select your OS and Python version adequately before downloading it.

Next step is to install the wheelhouse. We recommend that you create a dedicated Python environment for this (conda environment or pip virtual env, doesn't matter). In order to install all the packages inside the wheelhouse you can perform as follows:

```bash
unzip pyansys-v2023.2.10-all-wheelhouse-Windows-3.11.zip wheelhouse
pip install pyansys -f wheelhouse --no-index --upgrade --ignore-installed
```
This (line of code 2) should install all the dependencies inside the ``wheelhouse`` folder, where we extracted our ZIP file (line of code 1)

If you don't have internet access it might be a bit complicated. I would suggest that you provide access to PyPI in general (because there will be many calls to it).

**Reply by greschd** [reply]
> If you don't have internet access it might be a bit complicated. I would suggest that you provide access to PyPI in general (because there will be many calls to it).

@RobPasMue Since the wheelhouse (if I understand it correctly) includes all dependencies, shouldn't it be possible to install without internet access by passing the `--no-deps` flag to `pip`? Note that I haven't tested this.

Allowing all of `pypi.org` for internet access may be considered a fairly wide exception, since it contains all kinds of code.

**Reply by RobPasMue** [reply]
Hi @greschd! If you pass in the [``--no-deps`` flag](https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-no-deps) you will not be able to install the dependencies of the ``pyansys`` metapackage... which are basically all the PyAnsys libraries. Same happens for any other wheel you try to install using ``--no-deps``.

And I agree - allowing the entire ``pypi.org`` might be not recommended in terms of security. But getting a list of all the URLs to all our packages and their dependencies might be a never-ending task (new dependencies, removing old dependencies, new pyansys libraries etc.)

**Reply by greschd** [reply]
What I'm proposing is to install all packages in the wheelhouse, with a command like (powershell)

```
Get-ChildItem .\wheelhouse\ | ForEach-Object {python -m pip install --no-index --no-deps --upgrade --ignore-installed $_}
```
Equivalent bash syntax [1]:

```bash
python -m pip install --no-index --no-deps --upgrade --ignore-installed wheelhouse/*
```

My understanding is that the wheelhouse is exactly constructed to include all dependencies, so this will effectively result in the same install. [2]

[1] The bash command installs everything in one go whereas powershell uses one `pip install` per dependency, but that shouldn't matter.
[2] With one exception: The versions of all dependencies will be the one contained in the wheelhouse. The regular `pip install pyansys` install may have updated versions.

**Reply by greschd** [reply]
Ah sorry, I overlooked / misunderstood the `-f` flag in your original suggestion. I think that's the correct solution.

Why would you think that this should generate requests to PyPI? Since the wheelhouse is self-contained, it should work without internet access, no?

I just tried on my machine, and it seems to work when internet access is turned off.

**Reply by RobPasMue** [reply]
Yep, the ``-f`` option installs all the wheels in the wheelhouse.

> Why would you think that this should generate requests to PyPI? Since the wheelhouse is self-contained, it should work without internet access, no?

Precisely, this option does NOT generate requests to PyPI - since we are installing everything from the wheels. Accepting ``pypi.org`` was my proposal if they had some kind of limited access internet connection on the customer's machines and they wanted to install it from pip directly (i.e. not the wheelhouse)

**Reply by swparth** [reply]
Hi @RobPasMue and @greschd , Thank you so much for this instant discussion. A bit overwhelmed to see all the points. 

So between with internet and without internet, it seems without internet option would be more feasible because : 

1. It would be more secure
2. Getting a list of all the URLs to all our packages and their dependencies might be a never-ending task (new dependencies, removing old dependencies, new pyansys libraries etc.)

So now I focus completely on downloading the wheelhouse package and installing them. My python version is  3.11.4 and I downloaded pyansys-v2023.2.9-all-wheelhouse-Windows-3.11. 

1. Are these two compatible?
2.  Does the installation depend the location of individual folders ( of python and wheelhouse)?  

My next fundamental question is : You suggest to "create a dedicated Python environment for this". How do I do this, and why is this required?. If directly use unzip "C:\ProgramData\anaconda3\pyansys-v2023.2.9-all-wheelhouse-Windows-3.11.zip" wheelhouse i get an invalid syntax error.

**Reply by RobPasMue** [reply]
> So now I focus completely on downloading the wheelhouse package and installing them. My python version is 3.11.4 and I downloaded pyansys-v2023.2.9-all-wheelhouse-Windows-3.11.

Yes, that's right - If you have a Windows OS and Python 3.11.X that's your wheelhouse.

> If directly use unzip "C:\ProgramData\anaconda3\pyansys-v2023.2.9-all-wheelhouse-Windows-3.11.zip" wheelhouse I get an invalid syntax error.

Unzip the ZIP file like you would normally do in that case - there might be a mistake in my Powershell command. Go to the directory containing the unzipped folder and run:

```bash
pip install pyansys -f {your_unzipped_folder_name} --no-index --upgrade --ignore-installed
```

I suggest that you run the previous command with a new conda environment (since you are using conda). If you are not familiar with conda environments, have a look at their [documentation](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html)

**Reply by swparth** [reply]
Hi @RobPasMue 

So in my conda environment, i changed the path to the directory where wheelhouse packages are unzipped and then i ran your command. It seems like it was successful albeit some message as shown below :

```
DEPRECATION: ansys-dpf-core 0.8.1 has a non-standard dependency specifier ansys-dpf-gate>=0.3.*. pip 23.3 will enforce this behaviour change. A possible replacement is to upgrade to a newer version of ansys-dpf-core or contact the author to suggest that they release a version with a conforming dependency specifiers. Discussion can be found at https://github.com/pypa/pip/issues/12063
Installing collected packages: webencodings, pywin32, pytz, pypng, appdirs, zipp, websocket-client, urllib3, uritemplate, tzdata, typing-extensions, typing, sqlparse, six, setuptools, scooby, rfc3986, pyyaml, pywin32-ctypes, pyproject_hooks, pypiwin32, pyparsing, Pygments, pycparser, pyasn1, psutil, protobuf, plumbum, platformdirs, pkginfo, Pillow, packaging, numpy, more-itertools, mdurl, lxml, kiwisolver, idna, grpcio, fonttools, filelock, docutils, cycler, colorama, charset-normalizer, certifi, cachetools, asgiref, ansys-dpf-gatebin, tqdm, scipy, rsa, rpyc, requests, python-dateutil, pyasn1-modules, protoc-gen-swagger, markdown-it-py, jaraco.classes, importlib-metadata, httplib2, h5py, googleapis-common-protos, django, contourpy, click, cffi, build, bleach, ansys-api-systemcoupling, ansys-api-sherlock, ansys-api-pyensight, ansys-api-meshing-prime, ansys_api_mechanical, ansys-api-mapdl, ansys-api-fluent, rich, requests-toolbelt, requests-negotiate-sspi, readme-renderer, pyiges, pyansys-tools-versioning, pyansys-docker, pooch, pandas, matplotlib, keyring, grpcio-status, google-auth, cryptography, clr-loader, ansys-sherlock-core, ansys-seascape, ansys-optislang-core, ansys-motorcad-core, ansys-meshing-prime, ansys-api-platform-instancemanagement, vtk, twine, pytwin, pythonnet, pyspnego, google-auth-httplib2, google-api-core, ansys-systemcoupling-core, ansys-pythonnet, ansys-pyensight-core, ansys-platform-instancemanagement, requests-ntlm, pyvista, pyaedt, google-api-python-client, ansys-mechanical-core, ansys-fluent-core, ansys-dynamicreporting-core, ansys-openapi-common, ansys-mapdl-reader, ansys.grpc.dpf, ansys-mapdl-core, ansys-grantami-serverapi-openapi, ansys-grantami-bomanalytics-openapi, ansys-dpf-gate, ansys-math-core, ansys-grantami-recordlists, ansys-grantami-bomanalytics, ansys-dpf-core, ansys-dpf-post, ansys-dpf-composites, pyansys
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
conda-repo-cli 1.0.41 requires requests_mock, which is not installed.
gensim 4.3.0 requires FuzzyTM>=0.4.0, which is not installed.
tables 3.8.0 requires blosc2~=2.0.0, which is not installed.
tables 3.8.0 requires cython>=0.29.21, which is not installed.
transformers 2.1.1 requires sentencepiece, which is not installed.
aiohttp 3.8.3 requires charset-normalizer<3.0,>=2.0, but you have charset-normalizer 3.2.0 which is incompatible.
conda-repo-cli 1.0.41 requires clyent==1.2.1, but you have clyent 1.2.2 which is incompatible.
conda-repo-cli 1.0.41 requires nbformat==5.4.0, but you have nbformat 5.7.0 which is incompatible.
conda-repo-cli 1.0.41 requires PyYAML==6.0, but you have pyyaml 6.0.1 which is incompatible.
conda-repo-cli 1.0.41 requires requests==2.28.1, but you have requests 2.31.0 which is incompatible.
mdit-py-plugins 0.3.0 requires markdown-it-py<3.0.0,>=1.0.0, but you have markdown-it-py 3.0.0 which is incompatible.
numba 0.57.0 requires numpy<1.25,>=1.21, but you have numpy 1.25.2 which is incompatible.
python-lsp-black 1.2.1 requires black>=22.3.0, but you have black 0.0 which is incompatible.
sphinx 5.0.2 requires docutils<0.19,>=0.14, but you have docutils 0.20.1 which is incompatible.

Successfully installed Pillow-10.0.0 Pygments-2.15.1 ansys-api-fluent-0.3.12 ansys-api-mapdl-0.5.1 ansys-api-meshing-prime-0.1.1 ansys-api-platform-instancemanagement-1.0.0b3 ansys-api-pyensight-0.3.0 ansys-api-sherlock-0.1.14 ansys-api-systemcoupling-0.1.0 ansys-dpf-composites-0.2.0 ansys-dpf-core-0.8.1 ansys-dpf-gate-0.3.1 ansys-dpf-gatebin-0.4.1 ansys-dpf-post-0.4.0 ansys-dynamicreporting-core-0.4.0 ansys-fluent-core-0.14.0 ansys-grantami-bomanalytics-1.1.3 ansys-grantami-bomanalytics-openapi-1.0.0 ansys-grantami-recordlists-1.0.0 ansys-grantami-serverapi-openapi-1.0.0 ansys-mapdl-core-0.64.1 ansys-mapdl-reader-0.52.20 ansys-math-core-0.1.1 ansys-mechanical-core-0.8.0 ansys-meshing-prime-0.4.0 ansys-motorcad-core-0.1.3 ansys-openapi-common-1.2.1 ansys-optislang-core-0.2.1 ansys-platform-instancemanagement-1.1.1 ansys-pyensight-core-0.5.3 ansys-pythonnet-3.1.0rc1 ansys-seascape-0.2.0 ansys-sherlock-core-0.2.0 ansys-systemcoupling-core-0.1.3 ansys.grpc.dpf-0.8.1 ansys_api_mechanical-0.1.0 appdirs-1.4.4 asgiref-3.7.2 bleach-4.1.0 build-0.10.0 cachetools-5.3.1 certifi-2023.7.22 cffi-1.15.1 charset-normalizer-2.0.4 click-8.0.4 clr-loader-0.2.5 colorama-0.4.6 contourpy-1.0.5 cryptography-41.0.2 cycler-0.11.0 django-4.2.4 docutils-0.18.1 filelock-3.12.2 fonttools-4.25.0 google-api-core-2.11.1 google-api-python-client-2.95.0 google-auth-2.22.0 google-auth-httplib2-0.1.0 googleapis-common-protos-1.60.0 grpcio-1.56.2 grpcio-status-1.48.2 h5py-3.7.0 httplib2-0.22.0 idna-3.4 importlib-metadata-6.0.0 jaraco.classes-3.2.1 keyring-23.13.1 kiwisolver-1.4.4 lxml-4.9.2 markdown-it-py-2.2.0 matplotlib-3.7.1 mdurl-0.1.0 more-itertools-10.0.0 numpy-1.24.3 packaging-23.0 pandas-1.5.3 pkginfo-1.9.6 platformdirs-2.5.2 plumbum-1.8.2 pooch-1.4.0 protobuf-3.20.3 protoc-gen-swagger-0.1.0 psutil-5.9.0 pyaedt-0.6.76 pyansys-2023.2.9 pyansys-docker-5.0.4 pyansys-tools-versioning-0.4.0 pyasn1-0.4.8 pyasn1-modules-0.2.8 pycparser-2.21 pyiges-0.3.1 pyparsing-3.0.9 pypiwin32-223 pypng-0.20220715.0 pyproject_hooks-1.0.0 pyspnego-0.9.1 python-dateutil-2.8.2 pythonnet-3.0.1 pytwin-0.3.0 pytz-2022.7 pyvista-0.41.1 pywin32-305.1 pywin32-ctypes-0.2.0 pyyaml-6.0.1 readme-renderer-40.0 requests-2.31.0 requests-negotiate-sspi-0.5.2 requests-ntlm-1.2.0 requests-toolbelt-1.0.0 rfc3986-2.0.0 rich-13.5.2 rpyc-5.3.1 rsa-4.9 scipy-1.10.1 scooby-0.7.2 setuptools-68.0.0 six-1.16.0 sqlparse-0.4.4 tqdm-4.65.0 twine-4.0.2 typing-3.7.4.3 typing-extensions-4.7.1 tzdata-2023.3 uritemplate-4.1.1 urllib3-1.26.16 vtk-9.2.6 webencodings-0.5.1 websocket-client-0.58.0 zipp-3.11.0

```

So in my final pip list , what all packages should i be looking at so as to ensure all the files are installed successfully ?

**Reply by RobPasMue** [reply]
Hi @swparth - I think none of those dependencies are critical. My suggestion is that you start using pyansys libraries and see if any of them complain. It looks like your conda Python version might be missing some standard libraries. But that's just my guess.

**Reply by swparth** [reply]
Hi @RobPasMue , Yes, I went ahead and did that. However, my jupyter could not connect to the python kernel. So it threw this following statement : 

> `ImportError: version conflict: 'C:\\ProgramData\\anaconda3\\Lib\\site-packages\\psutil\\_psutil_windows.cp311-win_amd64.pyd' C extension module was built for another version of psutil (5.9.0 instead of 5.9.5); you may try to 'pip uninstall psutil', manually remove C:\ProgramData\anaconda3\Lib\site-packages\psutil\_psutil_windows.cp311-win_amd64.pyd or clean the virtual env somehow, then reinstall`

Is this because i did not use virtual environment to install?

**Reply by swparth** [reply]
I had one more comment. I check my pip list to see which all anysys package as installed : 

```
ansys-api-fluent                      0.3.12
ansys-api-mapdl                       0.5.1
ansys-api-mechanical                  0.1.0
ansys-api-meshing-prime               0.1.1
ansys-api-platform-instancemanagement 1.0.0b3
ansys-api-pyensight                   0.3.0
ansys-api-sherlock                    0.1.14
ansys-api-systemcoupling              0.1.0
ansys-dpf-composites                  0.2.0
ansys-dpf-core                        0.8.1
ansys-dpf-gate                        0.3.1
ansys-dpf-gatebin                     0.4.1
ansys-dpf-post                        0.4.0
ansys-dynamicreporting-core           0.4.0
ansys-fluent-core                     0.14.0
ansys-grantami-bomanalytics           1.1.3
ansys-grantami-bomanalytics-openapi   1.0.0
ansys-grantami-recordlists            1.0.0
ansys-grantami-serverapi-openapi      1.0.0
ansys-grpc-dpf                        0.8.1
ansys-mapdl-core                      0.64.1
ansys-mapdl-reader                    0.52.20
ansys-math-core                       0.1.1
ansys-mechanical-core                 0.8.0
ansys-meshing-prime                   0.4.0
ansys-motorcad-core                   0.1.3
ansys-openapi-common                  1.2.1
ansys-optislang-core                  0.2.1
ansys-platform-instancemanagement     1.1.1
ansys-pyensight-core                  0.5.3
ansys-pythonnet                       3.1.0rc1
ansys-seascape                        0.2.0
ansys-sherlock-core                   0.2.0
ansys-systemcoupling-core             0.1.3
```

I could not find pyfluent parametric and pyfluent visualization

**Reply by RobPasMue** [reply]
Some of the PyAnsys dependencies might be entering in conflict with your libraries. One option is that whenever you download the wheelhouse ZIP file, you also delete the wheel related to psutils inside it- if that's the one causing your issues locally.

And yes, looks like the extra dependencies are not being packaged properly. I will have to give it a look at our workflows - for some reason they are not being packaged.

**Reply by greschd** [reply]
> Yep, the -f option installs all the wheels in the wheelhouse.

My understanding is that it will just install the ones specified (here, `pyansys`) and its dependencies. Which means to get pyansys-parametric etc. the command should be

```
pip install pyansys[all] -f {your_unzipped_folder_name} --no-index --upgrade --ignore-installed
```
Note the added `[all]`.

But, this still leaves the problem that these packages are not actually present in the wheelhouse.

**Reply by RobPasMue** [reply]
Yes - thanks @greschd for reporting the issue. I just opened the PR to fix this https://github.com/ansys/actions/pull/343

But you are totally right, by default it won't install parametric or visualization. You need to add the all target.

**Reply by swparth** [reply]
> One option is that whenever you download the wheelhouse ZIP file, you also delete the wheel related to psutils inside it- if that's the one causing your issues locally.

I will be able to know this apriori or only in retrospect? It could be different for different systems?

Also can you tell me which version 'pip uninstall psutil' will uninstall? psutil (5.9.0 or 5.9.5) ? I should be having at-least one right?

**Reply by RobPasMue** [reply]
You should uninstall 5.9.5 from what I can see in your logs and have 5.9.0 installed. 

And it seems to me that you will only be able to know this in retrospective.

**Comment by swparth** [comment]
> pip install pyansys[all] -f {your_unzipped_folder_name} --no-index --upgrade --ignore-installed

this is what i get when I try to run the above command : 

ERROR: Cannot install pyansys[all]==2023.2.10 and pyansys[all]==2023.2.9 because these package versions have conflicting dependencies.

The conflict is caused by:
    pyansys[all] 2023.2.10 depends on ansys-meshing-prime==0.4.1
    pyansys[all] 2023.2.9 depends on ansys-mapdl-reader==0.52.13; extra == "all"

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts

**Reply by RobPasMue** [reply]
Yes, that's because you are installing different versions of the PyAnsys metapackage - and they are not compatible. If you want to install a new version of the PyAnsys metapackage (i.e. 2023.2.10) you have to start from scratch =)

**Reply by swparth** [reply]
Does starting from scratch mean uninstalling PyANSYS and reinstalling?  That's what i actually did. 
I ran the following :

```
 C:\ProgramData\anaconda3>pip uninstall pyansys
Found existing installation: pyansys 2023.2.9
Uninstalling pyansys-2023.2.9:
  Would remove:
    c:\programdata\anaconda3\lib\site-packages\pyansys-2023.2.9.dist-info\*
    c:\programdata\anaconda3\lib\site-packages\pyansys\*
Proceed (Y/n)? y
  Successfully uninstalled pyansys-2023.2.9

(base) C:\ProgramData\anaconda3>pip install pyansys[all] -f pyansys-v2023.2.9-all-wheelhouse-Windows-3.11 --no-index --upgrade --ignore-installed
```

**Reply by RobPasMue** [reply]
Yes but you only uninstalled PyAnsys and not all its dependencies. If you want to remove all the dependencies of a package, the only way I know is using the ``pip-autoremove`` package

```bash
pip install pip-autoremove
pip-autoremove pyansys -y
```

However, the fastest and cleanest way to do it would be to delete the existing virtual environment and create a new one

**Reply by swparth** [reply]
I think because of no internet, i'm unable to create a virtual environment: 

```
conda create --name pyansysenv 
CondaHTTPError: HTTP 000 CONNECTION FAILED for url <https://repo.anaconda.com/pkgs/main/win-64/current_repodata.json>
Elapsed: -

An HTTP error occurred when trying to retrieve this URL.
HTTP errors are often intermittent, and a simple retry will get you on your way.

If your current network has https://repo.anaconda.com blocked, please file
a support request with your network engineering team.

'https//repo.anaconda.com/pkgs/main/win-64' 
```
Should i halt my installation process till this bug (https://github.com/ansys/actions/pull/343) is resolved?

**Comment by swparth** [comment]
Also , i reinstalled conda (since i did not create virtual environment previously and auto-remove was not successful) tried pip install for pyansys (All)

```
C:\ProgramData\anaconda3>pip install pyansys[all] -f pyansys-v2023.2.10-all-wheelhouse-Windows-3.11 --no-index --upgrade --ignore-installed
Looking in links: pyansys-v2023.2.10-all-wheelhouse-Windows-3.11
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\pyansys-2023.2.10-py3-none-any.whl
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\ansys_mapdl_core-0.64.1-py3-none-any.whl (from pyansys[all])
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\ansys_dpf_core-0.8.1-py3-none-any.whl (from pyansys[all])
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\ansys_dpf_post-0.4.0-py3-none-any.whl (from pyansys[all])
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\ansys_dpf_gate-0.3.1-py3-none-any.whl (from pyansys[all])
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\ansys_dpf_composites-0.2.0-py3-none-any.whl (from pyansys[all])
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\ansys_dynamicreporting_core-0.4.0-py3-none-any.whl (from pyansys[all])
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\ansys_pyensight_core-0.5.3-py3-none-any.whl (from pyansys[all])
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\ansys_fluent_core-0.14.0-py3-none-any.whl (from pyansys[all])
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\pyaedt-0.6.76-py3-none-any.whl (from pyansys[all])
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\ansys_platform_instancemanagement-1.1.1-py3-none-any.whl (from pyansys[all])
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\ansys_grantami_bomanalytics-1.1.3-py3-none-any.whl (from pyansys[all])
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\ansys_grantami_recordlists-1.0.0-py3-none-any.whl (from pyansys[all])
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\ansys_openapi_common-1.2.1-py3-none-any.whl (from pyansys[all])
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\ansys_seascape-0.2.0-py3-none-any.whl (from pyansys[all])
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\ansys_sherlock_core-0.2.0-py3-none-any.whl (from pyansys[all])
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\ansys_meshing_prime-0.4.1-py3-none-any.whl (from pyansys[all])
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\pytwin-0.3.0-py3-none-any.whl (from pyansys[all])
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\ansys_systemcoupling_core-0.1.3-py3-none-any.whl (from pyansys[all])
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\ansys_motorcad_core-0.1.3-py3-none-any.whl (from pyansys[all])
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\ansys_math_core-0.1.1-py3-none-any.whl (from pyansys[all])
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\ansys_optislang_core-0.2.1-py3-none-any.whl (from pyansys[all])
Processing c:\programdata\anaconda3\pyansys-v2023.2.10-all-wheelhouse-windows-3.11\ansys_mechanical_core-0.8.0-py3-none-any.whl (from pyansys[all])
INFO: pip is looking at multiple versions of pyansys[all] to determine which version is compatible with other requirements. This could take a while.
ERROR: Could not find a version that satisfies the requirement ansys-mapdl-reader==0.52.13; extra == "all" (from pyansys[all]) (from versions: 0.52.20)
ERROR: No matching distribution found for ansys-mapdl-reader==0.52.13; extra == "all"
```

**Reply by RobPasMue** [reply]
HI @swparth - answering both https://github.com/ansys/pyansys/discussions/373#discussioncomment-7157590 and https://github.com/ansys/pyansys/discussions/373#discussioncomment-7157600

If you do not have an internet connection, running ``pyansys[all]`` with the existing wheelhouse ZIP file won't work, because it does not contain all the needed wheels. That is why you are getting this error:

```
ERROR: No matching distribution found for ansys-mapdl-reader==0.52.13; extra == "all"
```

I would suggest that you wait until we fix this issue - we would need to perform a new release of the PyAnsys metapackage. Please hold - I will inform you once it is ready.

**Reply by RobPasMue** [reply]
Hi @swparth - new release is out with the bugfix for the wheelhouse. These wheelhouse now contain all dependencies.

https://github.com/ansys/pyansys/releases/tag/v2023.2.11

**Comment by swparth** [comment]
Hi,

I tried a different work-around for this since, i was unable to create a virtual environment in a offline system. 
Both online/offline are windows system. 

**New strategy :**
1. Install conda and py3.11.4 in laptop (with internet access) , 
2. Create a virtual environment of the base conda packages.  (conda create --name py_blade --clone base)
3. Installation pyansys[all] inside the virtual environment.  (pip install pyansys[all]==2023.2.11)
4. Pack the conda file into a .tar.gz . (conda pack -n py_blade_pack)
5. Unpack this in the offline system.

However I see a problem with three packages during PYANSYS installation  : 
```
  Attempting uninstall: psutil
    Found existing installation: psutil 5.9.0
    Uninstalling psutil-5.9.0:
      Successfully uninstalled psutil-5.9.0
  Attempting uninstall: h5py
    Found existing installation: h5py 3.7.0
    Uninstalling h5py-3.7.0:
      Successfully uninstalled h5py-3.7.0
  Attempting uninstall: click
    Found existing installation: click 8.0.4
    Uninstalling click-8.0.4:
      Successfully uninstalled click-8.0.4
```

And then i pack my environment into conda pack :  
```
Files managed by conda were found to have been deleted/overwritten in the
following packages:
- click 8.0.4:
    lib\site-packages\click-8.0.4.dist-info\direct_url.json
    lib\site-packages\click-8.0.4.dist-info\installer
    lib\site-packages\click-8.0.4.dist-info\license.rst
    + 7 others
- h5py 3.7.0:
    lib\site-packages\h5py-3.7.0.dist-info\direct_url.json
    lib\site-packages\h5py-3.7.0.dist-info\installer
    lib\site-packages\h5py-3.7.0.dist-info\license
    + 5 others
- psutil 5.9.0:
    lib\site-packages\psutil-5.9.0.dist-info\direct_url.json
    lib\site-packages\psutil-5.9.0.dist-info\installer
    lib\site-packages\psutil-5.9.0.dist-info\license
    + 6 others
```

**Reply by swparth** [reply]
Hi @RobPasMue and @greschd. Thank you so much for the all the support. I really appreciate all the prompt response.
The above strategy worked for me.  PyANSYS has been depolyed in the offline system.
I want to close this discussion.

**Comment by whplace** [comment]
Hi,
I'm not really sure,
but ansys-mapdl-reader 0.53.0 seems to be unable to be installed in python3.8.

**Reply by RobPasMue** [reply]
Hi @whplace - in case you want to use ansys-mapdl-reader on Python 3.8, you need to build the package from scratch. Support for Python 3.8 was dropped in most of our PyAnsys libraries starting on October and following [NEP-0029](https://numpy.org/neps/nep-0029-deprecation_policy.html)
